libs/shared/flagd-core/src/lib/targeting/fractional.ts

```diff
@@ -1,20 +1,20 @@
-import { flagdPropertyKey, flagKeyPropertyKey, targetingPropertyKey } from './common';
 import MurmurHash3 from 'imurmurhash';
+import { flagKeyPropertyKey, flagdPropertyKey, targetingPropertyKey } from './common';
 
 export const fractionalRule = 'fractional';
 
-// eslint-disable-next-line  @typescript-eslint/no-explicit-any
-export function fractional(data: unknown, context: Record<any, any>): string | null {
-  if (!Array.isArray(data)) {
-    return null;
-  }
-
-  const args = Array.from(data);
+export function fractional(...args: unknown[]): string | null {
   if (args.length < 2) {
```

Comment:
This check should be done after you slice the array.

Reply:
right!